fvm/evm/emulator/emulator.go

```diff
@@ -244,10 +268,15 @@ func (proc *procedure) withdrawFrom(
 		return res, err
 	}
 
+	// if any error (invalid or vm) on the internal call, revert and don't commit any change
+	if res.Invalid() || res.Failed() {
+		return res, types.ErrInternalDirecCallFailed
```

Comment:
is it intentional to swallow the exact error? should we wrap it?

Reply:
It was intentional to stop the flow transaction, but now I think we might be able to send the result upward and do everything on the handler.

Reply:
I did more looking in it, the reason I think it might be a better option to make an intentional error for now is the direct call is also invisible to the user (we form it). we might revisit this in the future.